src/tiff.ml

```diff
@@ -960,9 +991,20 @@ module Data = struct
 end
 
 (* have to specify all 4 or else it defaults to whole file*)
-let data (type repr kind) ?window t (f : File.ro)
+let data (type repr kind) ?plane ?window t (f : File.ro)
     (data_type : (repr, kind) Data.kind) : (repr, kind) Data.t =
   let ifd = ifd t in
+  let planar_configuration = Ifd.planar_configuration ifd in
+  let plane =
+    match (planar_configuration, plane) with
+    | Planar, Some p -> p
+    | Planar, None -> 0
+    | Chunky, None -> 0
+    | _ ->
+        raise
```

Comment:
Can we split this into the two distinct errors (we can also use `invalid_arg` to save the bytes on `raise (Invalid_argument "")`)